components/admin/lessons/AdminLessonInputs.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react'
+import { AddModuleMutation, useAddModuleMutation } from '../../../graphql'
+import { formChange } from '../../../helpers/formChange'
+import { FormCard, MD_INPUT, TextField } from '../../FormCard'
+import { AlertFillIcon, CheckCircleIcon } from '@primer/octicons-react'
+import styles from '../../../scss/adminLessonInputs.module.scss'
+import { Spinner } from 'react-bootstrap'
+import { ApolloError } from 'apollo-server-micro'
+
+type Props = {
+  title?: string
+  lessonId: number
+  onAddModule?: (
+    m: AddModuleMutation['addModule'] | null,
+    e: { name: string; content: string } | null
+  ) => void
+}
+
+enum Error {
+  InvalidData = 'missing module name or description'
+}
+
+const values: [TextField, TextField] = [
+  {
+    title: 'Module Name',
+    value: ''
+  },
+  {
+    title: 'Description',
+    type: MD_INPUT,
+    value: ''
+  }
+]
+
+const AdminLessonInputs = ({ title, lessonId, onAddModule }: Props) => {
+  const [formOptions, setFormOptions] = useState<any>(values)
+  const [name, content] = formOptions
+
+  const [addModuleMutation, { data, error, loading }] = useAddModuleMutation({
+    variables: {
+      content: content.value || '',
+      lessonId,
+      name: name.value || ''
+    }
+  })
+
+  const [errorMsg, setErrorMsg] = useState(error?.message || '')
```

Comment:
This is not a PR blocker, but I noticed in @SlyBouhafs's PR he used lodash get instead of optional chaining. We should agree on which one to use for this codebase, or else inconsistencies will continue to appear

Reply:
One issue I've with lodash `get` is that it hinders code completion. I think the rule to use them should be as if statement and ternary operator.

- Use loadash `get` when the property is deeply nested. We don't want to do `obj?.obj?.obj?.obj || ''`.
- Use optional chaning when the property isn't deeply nested.

Reply:
What do you mean by `it hinders code completion`?

Both optional chaining and lodash get do the same thing though. We're using 2 different tools that do exactly the same thing, its like using 2 different API's that do exactly the same as each other.

Reply:
> What do you mean by it hinders code completion?

With lodash `get` we use a string `_.get('obj.obj')`. By using a string, the IDE/Extension/Plugin won't display auto-suggestions of the properties that this object has.

With string:
![image](https://user-images.githubusercontent.com/35906419/166135068-a94486fd-78f6-43d0-929d-ca28baa959ec.png)

Without:
![image](https://user-images.githubusercontent.com/35906419/166135098-20d66e3e-ecaf-4984-9af2-c93b42fe1ce1.png)